FileShareClient/FileShareApiClient.cs

```diff
@@ -23,12 +23,17 @@ public interface IFileShareApiClient
     public class FileShareApiClient : IFileShareApiClient
     {
         protected readonly IHttpClientFactory httpClientFactory;
+        protected readonly IAuthTokenProvider authTokenProvider;
+
+        public FileShareApiClient(IHttpClientFactory httpClientFactory, string baseAddress, IAuthTokenProvider authTokenProvider)
+        {
+            this.httpClientFactory = new SetBaseAddressHttpClientFactory(httpClientFactory, new Uri(baseAddress));
+            this.authTokenProvider = authTokenProvider;
+        }
 
         public FileShareApiClient(IHttpClientFactory httpClientFactory, string baseAddress, string accessToken)
+            :this(httpClientFactory, baseAddress, new DefaultAuthTokenProvider(accessToken))
         {
-            this.httpClientFactory = new AddAuthenticationHeaderHttpClientFactory(
```

Comment:
If the AddAuthenticationHeaderHttpClientFactory class is now redundant, then it probably should be deleted